templates/commands/goldentest/test_funcs.go

```diff
@@ -105,3 +113,71 @@ func parseTestConfig(path string) (*goldentest.Test, error) {
 	}
 	return test, nil
 }
+
+// getTempDir returns a temporary directory for golden tests usage.
+func getTempDir() (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", fmt.Errorf("failed to get home dir: %w", err)
+	}
+	tempDir := filepath.Join(
+		homeDir,
+		".abc",
+		"goldentests",
+		fmt.Sprint(time.Now().Unix()))
+
+	if err := os.MkdirAll(tempDir, ownerRWXPerms); err != nil {
+		return "", fmt.Errorf("MkdirAll(%q): %w", tempDir, err)
+	}
+	return tempDir, nil
+}
+
+// clearTestDir clears a test directory and only keeps the test config file.
+func clearTestDir(dir string) error {
+	files, err := os.ReadDir(dir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to read test dir: %w", err)
+	}
+
+	for _, file := range files {
+		if file.Name() != configName {
+			filePath := filepath.Join(dir, file.Name())
+			if err := os.RemoveAll(filePath); err != nil {
+				return fmt.Errorf("failed to remove outdated test artifact: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+// renderTestCase executes the "template render" command based upon test config.
+func renderTestCase(templateDir, outputDir string, tc *TestCase) error {
+	testDir := filepath.Join(outputDir, goldenTestDir, tc.TestName)
+
+	if err := clearTestDir(testDir); err != nil {
+		return fmt.Errorf("failed to clear test directory: %w", err)
+	}
+
+	args := []string{"--dest", testDir, "--force-overwrite"}
+	for _, input := range tc.TestConfig.Inputs {
+		args = append(args, "--input")
+		args = append(args, fmt.Sprintf("%s=%s", input.Name.Val, input.Value.Val))
+	}
+	args = append(args, templateDir)
+
+	r := &render.Command{}
```

Comment:
I'm not sure we should be using render.Command for this. Maybe we should factor out the rendering logic as a library, so we can call it like `render.Render(dest, location, inputs)`, without having to go through flags. Maybe this could be a TODO.

Reply:
I've leaved a TODO